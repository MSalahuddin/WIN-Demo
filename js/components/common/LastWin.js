import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

import Text, { SIZE, FONT_FAMILY } from './Text';
import { scale, scaleHeight } from '../../platformUtils';
import { LastWinIcon} from '../../../assets/images';
import { gameRoomStrings } from '../../stringConstants';
import { color } from '../../styles';

const Container = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
`;

const Prize = styled.Image`
  height: ${scaleHeight(24)};
  margin-right: ${scale(8)};
  width: ${scaleHeight(24)};
`;

const TextWrapper = styled.View`
  flex-direction: row;
`;

const LastWin = ({ timestamp, ...rest }) => {
  return (
    <Container {...rest}>
      <Prize source={LastWinIcon} />
      <TextWrapper>
        <Text size={SIZE.XSMALL}>{`${gameRoomStrings.lastWin} `}</Text>
        <Text color={color.silver} size={SIZE.XSMALL} fontFamily={FONT_FAMILY.REGULAR}>
          {timestamp}
        </Text>
      </TextWrapper>
    </Container>
  );
};

LastWin.propTypes = {
  timestamp: PropTypes.string
};

LastWin.defaultProps = {
  timestamp: ''
};

export default LastWin;
