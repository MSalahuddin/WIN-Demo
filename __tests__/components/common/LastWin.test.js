import React from 'react';
import { render } from 'react-native-testing-library';

import LastWin from '../../../js/components/common/LastWin';

describe('Component LastWin', () => {
  test('renders correctly', () => {
    const output = render(<LastWin timestamp="" />);
    expect(output).toMatchSnapshot();
  });
});
