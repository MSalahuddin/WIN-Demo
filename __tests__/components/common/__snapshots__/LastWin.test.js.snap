// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LastWin renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 48,
          "marginRight": 12,
          "width": 48,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginTop": 20,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Calibre-Medium",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
    >
      Last win: 
    </Text>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#000000"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "CalibreRegular",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
